crates/iota-types/src/event.rs

```diff
@@ -162,7 +168,13 @@ impl Event {
     }
 }
 
-// Event emitted in move code `fun advance_epoch`
+#[derive(Deserialize)]
+pub enum SystemEpochInfoEvent {
+    V1(SystemEpochInfoEventV1),
+    V2(SystemEpochInfoEventV2),
+}
+
+// Event emitted in move code `fun advance_epoch` in protocol versions 1 to 3
```

Comment:
Nit: line comments are hidden in the generated docs. Use doc comments instead for documenting the  API.
```suggestion
/// Event emitted in move code `fun advance_epoch` in protocol versions 1 to 3
```